test/test_unistd.rs

```diff
@@ -576,6 +577,59 @@ fn test_symlinkat() {
     );
 }
 
+
+#[test]
+#[should_panic (expected="`Err` value: Sys(EISDIR)")]
+fn test_unlinkat_dir_noremovedir() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let dirname = "foo_dir";
+    let dirpath = tempdir.path().join(dirname);
+
+    // Create dir
+    DirBuilder::new().recursive(true).create(&dirpath).unwrap();
+
+    // Get file descriptor for base directory
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+
+    // Attempt unlink dir at relative path without proper flag
+    unlinkat(Some(dirfd), dirname, UnlinkatFlags::NoRemoveDir).unwrap();
```

Comment:
You don't need `should_panic` for this.  Just replace the `unwrap` with an `unwrap_err` and check the result.